cli/cook/querying.py

```diff
@@ -190,19 +230,60 @@ def __get_latest_instance(job):
 
 
-def query_unique_and_run(clusters, job_or_instance_uuid, command_fn):
+def query_unique_and_run(clusters, entity_ref, command_fn):
     """Calls query_unique and then calls the given command_fn on the resulting job instance"""
-    query_result = query_unique(clusters, job_or_instance_uuid)
-    if query_result['type'] == 'job':
+    query_result = query_unique(clusters, entity_ref)
+    if query_result['type'] == Types.JOB:
         job = query_result['data']
         instance = __get_latest_instance(job)
         directory = mesos.retrieve_instance_sandbox_directory(instance, job)
         command_fn(instance, directory)
-    elif query_result['type'] == 'instance':
+    elif query_result['type'] == Types.INSTANCE:
         instance, job = query_result['data']
         directory = mesos.retrieve_instance_sandbox_directory(instance, job)
         command_fn(instance, directory)
     else:
         # This should not happen, because query_unique should
         # only return a map with type "job" or type "instance"
-        raise Exception(f'Encountered error when querying for {job_or_instance_uuid}.')
+        raise Exception(f'Encountered error when querying for {entity_ref}.')
+
+
+def parse_entity_refs(ref_strings):
+    """
+    Given a collection of entity ref strings, returns a list of entity ref maps, where each map has
+    the following shape:
+
+      {'cluster': ..., 'type': ..., 'uuid': ...}
+
+    The cluster field is the name of a cluster, the type field can be either job, instance, or group,
+    and the uuid field is the uuid of the entity in question. An entity ref string can either be
+    simply a UUID, in which case all configured clusters and all three types will be queried, or it
+    can be a /-delimited triple (cluster/type/uuid). The cluster and type portions can each be
+    specified as *, which means "all".
+
+    Some examples of valid entity ref strings follow:
+
+      5ab383b1-5b8b-4483-b2b2-45126923f4df
+      */*/5ab383b1-5b8b-4483-b2b2-45126923f4df
+      cluster1/*/5ab383b1-5b8b-4483-b2b2-45126923f4df
+      */job/5ab383b1-5b8b-4483-b2b2-45126923f4df
+      */instance/5ab383b1-5b8b-4483-b2b2-45126923f4df
+      */group/5ab383b1-5b8b-4483-b2b2-45126923f4df
+      cluster2/job/5ab383b1-5b8b-4483-b2b2-45126923f4df
+      cluster3/instance/5ab383b1-5b8b-4483-b2b2-45126923f4df
+      cluster4/group/5ab383b1-5b8b-4483-b2b2-45126923f4df
+
+    Throws if an invalid entity ref string is encountered.
+    """
+    entity_refs = []
+    for ref_string in ref_strings:
+        ref_string = ref_string.strip()
+        ref_parts = ref_string.split('/')
+        num_parts = len(ref_parts)
+        if num_parts == 1:
+            entity_refs.append({'cluster': Clusters.ALL, 'type': Types.ALL, 'uuid': ref_parts[0]})
+        elif num_parts == 3:
+            entity_refs.append({'cluster': unquote_plus(ref_parts[0]), 'type': ref_parts[1], 'uuid': ref_parts[2]})
```

Comment:
Why is the call to `unquote_plus` needed here?

Reply:
The idea is that the cluster name may have a `/` in it, in which case commands which output entity refs (in a future PR) will escape them in the cluster portion of the ref

Reply:
Based on our conversation just now, it sounds like we want to use `unquote` to avoid having any `+` characters in the cluster name changed to spaces.

Reply:
Done